### PR TITLE
[CI/Build] Update Ruff version

### PR DESCRIPTION
**Pull request by aarnphm**
Updated ruff to 0.6.5, possible to also run `ruff check`

Signed-off-by: Aaron Pham <contact@aarnphm.xyz>

---

<details>

<summary><b> PR Checklist (Click to Expand) </b></summary>

<p>Thank you for your contribution to vLLM! Before submitting the pull request, please ensure the PR meets the following criteria. This helps vLLM maintain the code quality and improve the efficiency of the review process.</p>

<h3>PR Title and Classification</h3>
<p>Only specific types of PRs will be reviewed. The PR title is prefixed appropriately to indicate the type of change. Please use one of the following:</p>
<ul>
    <li><code>[Bugfix]</code> for bug fixes.</li>
    <li><code>[CI/Build]</code> for build or continuous integration improvements.</li>
    <li><code>[Doc]</code> for documentation fixes and improvements.</li>
    <li><code>[Model]</code> for adding a new model or improving an existing model. Model name should appear in the title.</li>
    <li><code>[Frontend]</code> For changes on the vLLM frontend (e.g., OpenAI API server, <code>LLM</code> class, etc.) </li>
    <li><code>[Kernel]</code> for changes affecting CUDA kernels or other compute kernels.</li>
    <li><code>[Core]</code> for changes in the core vLLM logic (e.g., <code>LLMEngine</code>, <code>AsyncLLMEngine</code>, <code>Scheduler</code>, etc.)</li>
    <li><code>[Hardware][Vendor]</code> for hardware-specific changes. Vendor name should appear in the prefix (e.g., <code>[Hardware][AMD]</code>).</li>
    <li><code>[Misc]</code> for PRs that do not fit the above categories. Please use this sparingly.</li>
</ul>
<p><strong>Note:</strong> If the PR spans more than one category, please include all relevant prefixes.</p>

<h3>Code Quality</h3>

<p>The PR need to meet the following code quality standards:</p>

<ul>
    <li>We adhere to <a href="https://google.github.io/styleguide/pyguide.html">Google Python style guide</a> and <a href="https://google.github.io/styleguide/cppguide.html">Google C++ style guide</a>.</li>
    <li>Pass all linter checks. Please use <a href="https://github.com/vllm-project/vllm/blob/main/format.sh"><code>format.sh</code></a> to format your code.</li>
    <li>The code need to be well-documented to ensure future contributors can easily understand the code.</li>
    <li>Include sufficient tests to ensure the project to stay correct and robust. This includes both unit tests and integration tests.</li>
    <li>Please add documentation to <code>docs/source/</code> if the PR modifies the user-facing behaviors of vLLM. It helps vLLM user understand and utilize the new features or changes.</li>
</ul>

<h3>Adding or changing kernels</h3>
<p>Each custom kernel needs a schema and one or more implementations to be registered with PyTorch.</p>
<ul>
    <li>Make sure custom ops are registered following PyTorch guidelines: <a href="https://pytorch.org/tutorials/advanced/cpp_custom_ops.html#cpp-custom-ops-tutorial">Custom C++ and CUDA Operators</a> and <a href="https://docs.google.com/document/d/1_W62p8WJOQQUzPsJYa7s701JXt0qf2OfLub2sbkHOaU">The Custom Operators Manual</a></li>
    <li>Custom operations that return <code>Tensors</code> require meta-functions. Meta-functions should be implemented and registered in python so that dynamic dims can be handled automatically. See above documents for a description of meta-functions.</li>
    <li>Use <a href="https://pytorch.org/docs/stable/library.html#torch.library.opcheck"><code>torch.libary.opcheck()</code></a> to test the function registration and meta-function for any registered ops.  See <code>tests/kernels</code> for examples.</li>
    <li>When changing the C++ signature of an existing op, the schema must be updated to reflect the changes.</li>
    <li>If a new custom type is needed, see the following document: <a href="https://docs.google.com/document/d/18fBMPuOJ0fY5ZQ6YyrHUppw9FA332CpNtgB6SOIgyuA">Custom Class Support in PT2</a>.
</ul>

<h3>Notes for Large Changes</h3>
<p>Please keep the changes as concise as possible. For major architectural changes (>500 LOC excluding kernel/data/config/test), we would expect a GitHub issue (RFC) discussing the technical design and justification. Otherwise, we will tag it with <code>rfc-required</code> and might not go through the PR.</p>

<h3>What to Expect for the Reviews</h3>

<p>The goal of the vLLM team is to be a <i>transparent reviewing machine</i>. We would like to make the review process transparent and efficient and make sure no contributor feel confused or frustrated. However, the vLLM team is small, so we need to prioritize some PRs over others. Here is what you can expect from the review process: </p>

<ul>
    <li> After the PR is submitted, the PR will be assigned to a reviewer. Every reviewer will pick up the PRs based on their expertise and availability.</li>
    <li> After the PR is assigned, the reviewer will provide status update every 2-3 days. If the PR is not reviewed within 7 days, please feel free to ping the reviewer or the vLLM team.</li>
    <li> After the review, the reviewer will put an <code> action-required</code> label on the PR if there are changes required. The contributor should address the comments and ping the reviewer to re-review the PR.</li>
    <li> Please respond to all comments within a reasonable time frame. If a comment isn't clear or you disagree with a suggestion, feel free to ask for clarification or discuss the suggestion.
 </li>
</ul>

<h3>Thank You</h3>

<p> Finally, thank you for taking the time to read these guidelines and for your interest in contributing to vLLM. Your contributions make vLLM a great tool for everyone! </p>

</details>

### CODE REVIEW

**Comment by DarkLight1337** [comment]
We need to update the CI to use `ruff check`

**Comment by DarkLight1337** [comment]
I would refrain from adding pyupgrade in this PR since it looks like there are already enough changes as it stands just from upgrading ruff.

**Comment by DarkLight1337** [comment]
Also, please merge from main as #7820 updated quite a few files.

**Comment by DarkLight1337** [comment]
I have to go now so take your time to refine the lint ruleset.

**Comment by aarnphm** [comment]
> I have to go now so take your time to refine the lint ruleset.

thanks a lot

File: tests/test_logger.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ def test_an_error_is_raised_when_custom_logging_config_file_does_not_exist():
     configuration occurs."""
     with pytest.raises(RuntimeError) as ex_info:
         _configure_vllm_root_logger()
-    assert ex_info.type == RuntimeError
```
Comment (DarkLight1337) [inline tests/test_logger.py:114]: There are some cases where we intentionally check the exact types match. Can you ignore all existing cases so they aren't modified by ruff?

File: pyproject.toml [hunk 2]
```diff
@@ -12,8 +12,8 @@ requires = [
 build-backend = "setuptools.build_meta"
 
 [tool.ruff]
-# Allow lines to be as long as 80.
-line-length = 80
+# Allow lines to be as long as 119
+line-length = 119
```
Comment (DarkLight1337) [inline pyproject.toml:16]: I believe yapf enforces a line length of 80, so let's not change that.

File: pyproject.toml [hunk 3]
```diff
@@ -29,8 +29,6 @@ select = [
     # "UP",
     # flake8-bugbear
     "B",
-    # flake8-simplify
-    "SIM",
```
Comment (DarkLight1337) [inline pyproject.toml:33]: Why is this removed?

Comment (aarnphm) [reply pyproject.toml:33]: Because I think it introduces a lot of changes, and it will just make the diff a lot larger

File: tests/test_logger.py (hunk 1 above)
Comment (aarnphm) [reply tests/test_logger.py:114]: wouldn't isinstance does this?

File: pyproject.toml (hunk 3 above)
Comment (DarkLight1337) [reply pyproject.toml:33]: Can we keep the rules that were there in the original ruff version?

File: tests/test_logger.py (hunk 1 above)
Comment (DarkLight1337) [reply tests/test_logger.py:114]: No, `isinstance` is less strict as subclasses would also match.

File: pyproject.toml (hunk 3 above)
Comment (aarnphm) [reply pyproject.toml]: sg

File: tests/test_logger.py (hunk 1 above)
Comment (DarkLight1337) [reply tests/test_logger.py:114]: Also, this change causes `assert_nested_tensors_equal` to fail as shown in the CI logs.

File: vllm/attention/backends/utils.py [hunk 4]
```diff
@@ -33,10 +33,9 @@ def is_block_tables_empty(block_tables: Union[None, Dict]):
     """
     if block_tables is None:
         return True
-    if isinstance(block_tables, dict) and all(
-            value is None for value in block_tables.values()):
-        return True
-    return False
+    return bool(
+        isinstance(block_tables, dict)
+        and all(value is None for value in block_tables.values()))
```
Comment (DarkLight1337) [inline vllm/attention/backends/utils.py:38]:
```suggestion
    return (
        isinstance(block_tables, dict)
        and all(value is None for value in block_tables.values())
    )
```

File: vllm/attention/ops/triton_flash_attention.py [hunk 5]
```diff
@@ -139,7 +139,7 @@ def _attn_fwd_inner(
                 size_n = start_n + OFFS_N[None, :]
                 mask = size_n < boundary_m[:, None]
                 qk = tl.where(mask, qk, float("-inf"))
-        if IS_CAUSAL:
+        if IS_CAUSAL:  # noqa: SIM102
```
Comment (DarkLight1337) [inline vllm/attention/ops/triton_flash_attention.py:142]: Is this required?

File: vllm/engine/async_llm_engine.py [hunk 6]
```diff
@@ -806,7 +806,7 @@ async def generate(
             request_id: The unique id of the request.
             lora_request: LoRA request to use for generation, if any.
             trace_headers: OpenTelemetry trace headers.
-            prompt_adapter_request: Prompt Adapter request to use 
+            prompt_adapter_request: Prompt Adapter request to use
                                             for generation, if any.
```
Comment (DarkLight1337) [inline vllm/engine/async_llm_engine.py:810]:
```suggestion
            prompt_adapter_request: Prompt Adapter request to use
                for generation, if any.
```

File: vllm/engine/async_llm_engine.py [hunk 7]
```diff
@@ -1022,15 +1022,15 @@ def remove_logger(self, logger_name: str) -> None:
     async def start_profile(self) -> None:
         # using type instead of isinstance to check to avoid capturing
         # inherited classes
-        if type(self.engine.model_executor) == GPUExecutorAsync:
```
Comment (DarkLight1337) [inline vllm/engine/async_llm_engine.py:1025]: Don't use `isinstance` checks here either. See the comment for this line

File: vllm/model_executor/guided_decoding/outlines_logits_processors.py [hunk 8]
```diff
@@ -67,9 +67,9 @@ def __call__(self, input_ids: List[int],
         instruction = self._guide.get_next_instruction(
             state=self._fsm_state[seq_id])
 
-        if type(instruction) == Generate:
+        if isinstance(instruction, Generate):
             allowed_tokens = instruction.tokens
-        elif type(instruction) == Write:
```
Comment (DarkLight1337) [inline vllm/model_executor/guided_decoding/outlines_logits_processors.py:72]: As above

File: vllm/triton_utils/libentry.py [hunk 9]
```diff
@@ -35,8 +35,8 @@ def key(self, spec_args, dns_args, const_args):
         dns_key = [
             arg.dtype if hasattr(
                 arg, "data_ptr") else type(arg) if not isinstance(arg, int)
-            else "i32" if -(2**31) <= arg and arg <= 2**31 -
-            1 else "u64" if 2**63 <= arg and arg <= 2**64 - 1 else "i64"
+            else "i32" if arg >= -(2**31) and arg <= 2**31 -
+            1 else "u64" if arg >= 2**63 and arg <= 2**64 - 1 else "i64"
```
Comment (DarkLight1337) [inline vllm/triton_utils/libentry.py:39]:
```suggestion
            else "i32" if -(2**31) <= arg <= 2**31 - 1
            else "u64" if 2**63 <= arg <= 2**64 - 1 else "i64"
```

Also I hate this chained ternary statement...

**Review by dtrifiro — APPROVED** [review]

File: vllm/engine/async_llm_engine.py (hunk 6 above)
Comment (aarnphm) [reply vllm/engine/async_llm_engine.py:810]: hmm, idk why ruff format this 🤔

File: vllm/attention/backends/utils.py (hunk 4 above)
Comment (aarnphm) [reply vllm/attention/backends/utils.py:38]: huh strange, since ruff format run this

File: vllm/engine/async_llm_engine.py (hunk 6 above)
Comment (DarkLight1337) [reply vllm/engine/async_llm_engine.py:810]: Which rule is this under?

File: vllm/attention/backends/utils.py (hunk 4 above)
Comment (DarkLight1337) [reply vllm/attention/backends/utils.py:38]: Which rule is this under?

File: vllm/engine/async_llm_engine.py (hunk 6 above)
Comment (aarnphm) [reply vllm/engine/async_llm_engine.py:810]: This is their rules afaict. There aren't ways to ignore this other than `# ruff: format: ignore`

Comment (DarkLight1337) [reply vllm/engine/async_llm_engine.py:810]: Ok let's just follow their rules then. No big deal as long as the docstring still appears correctly.

File: vllm/triton_utils/libentry.py (hunk 9 above)
Comment (aarnphm) [reply vllm/triton_utils/libentry.py:39]: Let's refactor this outside of this PR?

**Comment by aarnphm** [comment]
sorry for the late response, but I think this should be ready now?

File: vllm/attention/backends/utils.py (hunk 4 above)
Comment (DarkLight1337) [reply vllm/attention/backends/utils.py:38]: Any update on this?

File: vllm/spec_decode/metrics.py [hunk 10]
```diff
@@ -104,13 +104,10 @@ def _should_collect_rejsample_metrics(self, now: float) -> bool:
         if self._rank != 0:
             return False
 
-        if (now - self._last_metrics_collect_time <
-                self._rejsample_metrics_collect_interval_s):
-            return False
-        return True
+        return not now - self._last_metrics_collect_time < self._rejsample_metrics_collect_interval_s  # noqa: E501
```
Comment (DarkLight1337) [inline vllm/spec_decode/metrics.py:107]:
```suggestion
        return now - self._last_metrics_collect_time >= self._rejsample_metrics_collect_interval_s  # noqa: E501
```

File: vllm/triton_utils/libentry.py (hunk 9 above)
Comment (DarkLight1337) [reply vllm/triton_utils/libentry.py:39]: Sure!

File: vllm/attention/ops/triton_flash_attention.py (hunk 5 above)
Comment (aarnphm) [reply vllm/attention/ops/triton_flash_attention.py]: nope.

File: vllm/attention/backends/utils.py (hunk 4 above)
Comment (aarnphm) [reply vllm/attention/backends/utils.py:38]: This is just `ruff format` afaict

File: vllm/attention/ops/triton_flash_attention.py (hunk 5 above)
Comment (DarkLight1337) [reply vllm/attention/ops/triton_flash_attention.py]: Let's remove the `noqa` then.

File: vllm/attention/backends/utils.py (hunk 4 above)
Comment (DarkLight1337) [reply vllm/attention/backends/utils.py]: Currently, we should use `yapf` instead of `ruff` for formatting. So you can apply my suggestion unless `ruff check` raises an error.

File: vllm/attention/ops/triton_flash_attention.py (hunk 5 above)
Comment (aarnphm) [reply vllm/attention/ops/triton_flash_attention.py]: I think this is already addressed at latest change.

Comment (DarkLight1337) [reply vllm/attention/ops/triton_flash_attention.py]: You're right, never mind then.

**Review by DarkLight1337 — APPROVED** [review]
LGTM now. However there is currently some problem with CI, so it may take longer to get this merged.

**Comment by DarkLight1337** [comment]
Thanks for your time and effort!

**Comment by aarnphm** [comment]
Hmm is there a way to retrigger buildkite runs?

**Comment by DarkLight1337** [comment]
I can retrigger them but it won't help until the underlying cause is fixed first.

**Comment by aarnphm** [comment]
Seems like there are problems with AWS. anw thanks again for all the help
